### DISCUSSION TITLE
How to pass display through code-server docker edition?

### DISCUSSION THREAD

**Discussion by tydaytygx** [coder/code-server#4730] [Q&A]
For example I want to use turtle **GUI** with Python3
```python
from turtle import *
```
Could I use browser to show the GUI display?
I got python3-tk installed but it shows the error like this
```log
_tkinter.TclError: no display name and no $DISPLAY environment variable
```

**Comment by lg-nineoneninethreenine** [comment]
You need noVNC to access that. @jsjoeio can you try checking if you can install noVNC on the docker editions?

**Reply by jsjoeio** [reply]
I am not familiar with noVNC so I'm not sure I can help much here! 

If you're using Docker though, couldn't you add the environment variables needed to use turtle to your Dockerfile?

**Reply by tydaytygx** [reply]
> 
Thanks for the reply. I would try.

**Reply by tydaytygx** [reply]
> I am not familiar with noVNC so I'm not sure I can help much here!
> 
> If you're using Docker though, couldn't you add the environment variables needed to use turtle to your Dockerfile?

I run the docker container on a non-GUI linux server, should I create a virtual display to forward the GUI?

**Reply by lg-nineoneninethreenine** [reply]
Afaik, I use Gitpod and they kinda use containers. Their Docker images with noVNC(source code [here](https://github.com/gitpod/workspace-images)) use x11vnc, Openbox and some code to put it together. However, I wonder if it'd work here too.

**Reply by jsjoeio** [reply]
> I run the docker container on a non-GUI linux server, should I create a virtual display to forward the GUI?

maybe? I'm not super familiar with virtual displays so I can't help much here :(
